storage/innobase/include/dict0mem.h

```diff
@@ -653,18 +653,28 @@ struct dict_col_t{
 	bool is_nullable() const { return !(prtype & DATA_NOT_NULL); }
 
 	/** @return whether this is system field */
-	bool vers_sys_field() const { return prtype & DATA_VERSIONED; }
+	bool vers_sys_field() const
+	{
+		return vers_sys_start() || vers_sys_end();
+	}
+	/** @return whether table of this system field is TRX_ID-based */
+	bool vers_native() const
+	{
+		ut_ad(vers_sys_field());
+		ut_ad(mtype == DATA_INT || mtype == DATA_FIXBINARY);
+		return mtype == DATA_INT;
+	}
```

Comment:
I think that it is OK to add the new member function `vers_native()`, but I oppose the change to `vers_sys_field()`.